apps/extension/src/app/layout.tsx

```diff
@@ -1,17 +1,94 @@
+import { Button } from "@agreeto/ui";
+import { useEffect } from "react";
+
 import { Navbar } from "~app/navbar";
+import { useValidateTrialOrSub } from "~features/auth/is-authed";
+import { trpcApi } from "~features/trpc/api/hooks";
 
 export const Layout: React.FC<{ children?: React.ReactNode }> = ({
   children,
 }) => {
+  // Check that trial/subscription are valid
+  const renderComponent = useValidateTrialOrSub();
+
+  const utils = trpcApi.useContext();
+  useEffect(() => {
+    // prefetch subscription
+    utils.user.subscription.prefetch();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Curious to me that you call this prefetch from the component and not the router. 🤔